// @uiw/codemirror-theme-eclipse/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.eclipseInit = exports.eclipse = void 0;
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _highlight = require("@lezer/highlight");
var _codemirrorThemes = require("@uiw/codemirror-themes");
var eclipseInit = function eclipseInit(options) {
  var _ref = options || {},
    _ref$theme = _ref.theme,
    theme = _ref$theme === void 0 ? 'light' : _ref$theme,
    _ref$settings = _ref.settings,
    settings = _ref$settings === void 0 ? {} : _ref$settings,
    _ref$styles = _ref.styles,
    styles = _ref$styles === void 0 ? [] : _ref$styles;
  return (0, _codemirrorThemes.createTheme)({
    theme: theme,
    settings: (0, _objectSpread2["default"])({
      background: '#fff',
      foreground: '#000',
      caret: '#FFFFFF',
      selection: '#d7d4f0',
      selectionMatch: '#d7d4f0',
      gutterBackground: '#f7f7f7',
      gutterForeground: '#999',
      lineHighlight: '#e8f2ff',
      gutterBorder: 'transparent'
    }, settings),
    styles: [{
      tag: [_highlight.tags.comment],
      color: '#3F7F5F'
    }, {
      tag: [_highlight.tags.documentMeta],
      color: '#FF1717'
    }, {
      tag: _highlight.tags.keyword,
      color: '#7F0055',
      fontWeight: 'bold'
    }, {
      tag: _highlight.tags.atom,
      color: '#00f'
    }, {
      tag: _highlight.tags.number,
      color: '#164'
    }, {
      tag: _highlight.tags.propertyName,
      color: '#164'
    }, {
      tag: [_highlight.tags.variableName, _highlight.tags.definition(_highlight.tags.variableName)],
      color: '#0000C0'
    }, {
      tag: _highlight.tags["function"](_highlight.tags.variableName),
      color: '#0000C0'
    }, {
      tag: _highlight.tags.string,
      color: '#2A00FF'
    }, {
      tag: _highlight.tags.operator,
      color: 'black'
    }, {
      tag: _highlight.tags.tagName,
      color: '#170'
    }, {
      tag: _highlight.tags.attributeName,
      color: '#00c'
    }, {
      tag: _highlight.tags.link,
      color: '#219'
    }].concat((0, _toConsumableArray2["default"])(styles))
  });
};
exports.eclipseInit = eclipseInit;
var eclipse = eclipseInit();
exports.eclipse = eclipse;
