// @uiw/codemirror-theme-abcdef/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.abcdefInit = exports.abcdef = void 0;
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _highlight = require("@lezer/highlight");
var _codemirrorThemes = require("@uiw/codemirror-themes");
/**
 * @name abcdef
 * @author codemirror.net
 * https://codemirror.net/5/theme/abcdef.css
 */

var abcdefInit = function abcdefInit(options) {
  var _ref = options || {},
    _ref$theme = _ref.theme,
    theme = _ref$theme === void 0 ? 'dark' : _ref$theme,
    _ref$settings = _ref.settings,
    settings = _ref$settings === void 0 ? {} : _ref$settings,
    _ref$styles = _ref.styles,
    styles = _ref$styles === void 0 ? [] : _ref$styles;
  return (0, _codemirrorThemes.createTheme)({
    theme: theme,
    settings: (0, _objectSpread2["default"])({
      background: '#0f0f0f',
      foreground: '#defdef',
      caret: '#00FF00',
      selection: '#515151',
      selectionMatch: '#515151',
      gutterBackground: '#555',
      gutterForeground: '#FFFFFF',
      lineHighlight: '#314151'
    }, settings),
    styles: [{
      tag: _highlight.tags.keyword,
      color: 'darkgoldenrod',
      fontWeight: 'bold'
    }, {
      tag: _highlight.tags.atom,
      color: '#77F'
    }, {
      tag: _highlight.tags.comment,
      color: '#7a7b7c',
      fontStyle: 'italic'
    }, {
      tag: _highlight.tags.number,
      color: 'violet'
    }, {
      tag: _highlight.tags.definition(_highlight.tags.variableName),
      color: '#fffabc'
    }, {
      tag: _highlight.tags.variableName,
      color: '#abcdef'
    }, {
      tag: _highlight.tags["function"](_highlight.tags.variableName),
      color: '#fffabc'
    }, {
      tag: _highlight.tags.typeName,
      color: '#FFDD44'
    }, {
      tag: _highlight.tags.tagName,
      color: '#def'
    }, {
      tag: _highlight.tags.string,
      color: '#2b4'
    }, {
      tag: _highlight.tags.meta,
      color: '#C9F'
    },
    // { tag: t.qualifier, color: '#FFF700' },
    // { tag: t.builtin, color: '#30aabc' },
    {
      tag: _highlight.tags.bracket,
      color: '#8a8a8a'
    }, {
      tag: _highlight.tags.attributeName,
      color: '#DDFF00'
    }, {
      tag: _highlight.tags.heading,
      color: 'aquamarine',
      fontWeight: 'bold'
    }, {
      tag: _highlight.tags.link,
      color: 'blueviolet',
      fontWeight: 'bold'
    }].concat((0, _toConsumableArray2["default"])(styles))
  });
};
exports.abcdefInit = abcdefInit;
var abcdef = abcdefInit();
exports.abcdef = abcdef;
