// @uiw/codemirror-theme-androidstudio/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.androidstudioInit = exports.androidstudio = void 0;
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _highlight = require("@lezer/highlight");
var _codemirrorThemes = require("@uiw/codemirror-themes");
/**
 * @name androidstudio
 */

var androidstudioInit = function androidstudioInit(options) {
  var _ref = options || {},
    _ref$theme = _ref.theme,
    theme = _ref$theme === void 0 ? 'dark' : _ref$theme,
    _ref$settings = _ref.settings,
    settings = _ref$settings === void 0 ? {} : _ref$settings,
    _ref$styles = _ref.styles,
    styles = _ref$styles === void 0 ? [] : _ref$styles;
  return (0, _codemirrorThemes.createTheme)({
    theme: theme,
    settings: (0, _objectSpread2["default"])({
      background: '#282b2e',
      foreground: '#a9b7c6',
      caret: '#00FF00',
      selection: '#343739',
      selectionMatch: '#343739',
      lineHighlight: '#343739'
    }, settings),
    styles: [{
      tag: [_highlight.tags.keyword, _highlight.tags.deleted, _highlight.tags.className],
      color: '#cc7832'
    }, {
      tag: [_highlight.tags.number, _highlight.tags.literal, _highlight.tags.derefOperator],
      color: '#6897bb'
    }, {
      tag: [_highlight.tags.link, _highlight.tags.variableName],
      color: '#629755'
    }, {
      tag: [_highlight.tags.comment, _highlight.tags.quote],
      color: 'grey'
    }, {
      tag: [_highlight.tags.meta, _highlight.tags.documentMeta],
      color: '#bbb529'
    }, {
      tag: [_highlight.tags.string, _highlight.tags.propertyName, _highlight.tags.attributeValue],
      color: '#6a8759'
    }, {
      tag: [_highlight.tags.heading, _highlight.tags.typeName],
      color: '#ffc66d'
    }, {
      tag: [_highlight.tags.attributeName],
      color: '#a9b7c6'
    }, {
      tag: [_highlight.tags.emphasis],
      fontStyle: 'italic'
    }].concat((0, _toConsumableArray2["default"])(styles))
  });
};
exports.androidstudioInit = androidstudioInit;
var androidstudio = androidstudioInit();
exports.androidstudio = androidstudio;
