// @uiw/codemirror-theme-noctis-lilac/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.noctisLilacInit = exports.noctisLilac = void 0;
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _highlight = require("@lezer/highlight");
var _codemirrorThemes = require("@uiw/codemirror-themes");
var noctisLilacInit = function noctisLilacInit(options) {
  var _ref = options || {},
    _ref$theme = _ref.theme,
    theme = _ref$theme === void 0 ? 'light' : _ref$theme,
    _ref$settings = _ref.settings,
    settings = _ref$settings === void 0 ? {} : _ref$settings,
    _ref$styles = _ref.styles,
    styles = _ref$styles === void 0 ? [] : _ref$styles;
  return (0, _codemirrorThemes.createTheme)({
    theme: theme,
    settings: (0, _objectSpread2["default"])({
      background: '#f2f1f8',
      foreground: '#0c006b',
      caret: '#5c49e9',
      selection: '#d5d1f2',
      selectionMatch: '#d5d1f2',
      gutterBackground: '#f2f1f8',
      gutterForeground: '#0c006b70',
      lineHighlight: '#e1def3'
    }, settings),
    styles: [{
      tag: _highlight.tags.comment,
      color: '#9995b7'
    }, {
      tag: _highlight.tags.keyword,
      color: '#ff5792',
      fontWeight: 'bold'
    }, {
      tag: [_highlight.tags.definitionKeyword, _highlight.tags.modifier],
      color: '#ff5792'
    }, {
      tag: [_highlight.tags.className, _highlight.tags.tagName, _highlight.tags.definition(_highlight.tags.typeName)],
      color: '#0094f0'
    }, {
      tag: [_highlight.tags.number, _highlight.tags.bool, _highlight.tags["null"], _highlight.tags.special(_highlight.tags.brace)],
      color: '#5842ff'
    }, {
      tag: [_highlight.tags.definition(_highlight.tags.propertyName), _highlight.tags["function"](_highlight.tags.variableName)],
      color: '#0095a8'
    }, {
      tag: _highlight.tags.typeName,
      color: '#b3694d'
    }, {
      tag: [_highlight.tags.propertyName, _highlight.tags.variableName],
      color: '#fa8900'
    }, {
      tag: _highlight.tags.operator,
      color: '#ff5792'
    }, {
      tag: _highlight.tags.self,
      color: '#e64100'
    }, {
      tag: [_highlight.tags.string, _highlight.tags.regexp],
      color: '#00b368'
    }, {
      tag: [_highlight.tags.paren, _highlight.tags.bracket],
      color: '#0431fa'
    }, {
      tag: _highlight.tags.labelName,
      color: '#00bdd6'
    }, {
      tag: _highlight.tags.attributeName,
      color: '#e64100'
    }, {
      tag: _highlight.tags.angleBracket,
      color: '#9995b7'
    }].concat((0, _toConsumableArray2["default"])(styles))
  });
};
exports.noctisLilacInit = noctisLilacInit;
var noctisLilac = noctisLilacInit();
exports.noctisLilac = noctisLilac;
