// @uiw/codemirror-theme-darcula/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.darculaInit = exports.darcula = void 0;
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _highlight = require("@lezer/highlight");
var _codemirrorThemes = require("@uiw/codemirror-themes");
/**
 * @name darcula
 * @author darcula
 * Name: IntelliJ IDEA darcula theme
 * From IntelliJ IDEA by JetBrains
 */

var darculaInit = function darculaInit(options) {
  var _ref = options || {},
    _ref$theme = _ref.theme,
    theme = _ref$theme === void 0 ? 'light' : _ref$theme,
    _ref$settings = _ref.settings,
    settings = _ref$settings === void 0 ? {} : _ref$settings,
    _ref$styles = _ref.styles,
    styles = _ref$styles === void 0 ? [] : _ref$styles;
  return (0, _codemirrorThemes.createTheme)({
    theme: theme,
    settings: (0, _objectSpread2["default"])({
      background: '#2B2B2B',
      foreground: '#f8f8f2',
      caret: '#FFFFFF',
      selection: 'rgba(255, 255, 255, 0.1)',
      selectionMatch: 'rgba(255, 255, 255, 0.2)',
      gutterBackground: 'rgba(255, 255, 255, 0.1)',
      gutterForeground: '#999',
      gutterBorder: 'transparent',
      lineHighlight: 'rgba(255, 255, 255, 0.1)'
    }, settings),
    styles: [{
      tag: [_highlight.tags.atom, _highlight.tags.number],
      color: '#bd93f9'
    }, {
      tag: [_highlight.tags.comment],
      color: '#61A151'
    }, {
      tag: [_highlight.tags.string],
      color: '#6A8759'
    }, {
      tag: [_highlight.tags.variableName, _highlight.tags.operator],
      color: '#A9B7C6'
    }, {
      tag: [_highlight.tags.meta, _highlight.tags.className],
      color: '#A9B7C6'
    }, {
      tag: [_highlight.tags.propertyName],
      color: '#FFC66D'
    }, {
      tag: [_highlight.tags.keyword],
      color: '#CC7832'
    }, {
      tag: [_highlight.tags.tagName],
      color: '#ff79c6'
    }, {
      tag: [_highlight.tags.typeName],
      color: '#ffb86c'
    }].concat((0, _toConsumableArray2["default"])(styles))
  });
};
exports.darculaInit = darculaInit;
var darcula = darculaInit();
exports.darcula = darcula;
