// @uiw/codemirror-theme-bbedit/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.bbeditInit = exports.bbedit = void 0;
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _highlight = require("@lezer/highlight");
var _codemirrorThemes = require("@uiw/codemirror-themes");
var bbeditInit = function bbeditInit(options) {
  var _ref = options || {},
    _ref$theme = _ref.theme,
    theme = _ref$theme === void 0 ? 'light' : _ref$theme,
    _ref$settings = _ref.settings,
    settings = _ref$settings === void 0 ? {} : _ref$settings,
    _ref$styles = _ref.styles,
    styles = _ref$styles === void 0 ? [] : _ref$styles;
  return (0, _codemirrorThemes.createTheme)({
    theme: theme,
    settings: (0, _objectSpread2["default"])({
      background: '#FFFFFF',
      foreground: '#000000',
      caret: '#FBAC52',
      selection: '#FFD420',
      selectionMatch: '#FFD420',
      gutterBackground: '#f5f5f5',
      gutterForeground: '#4D4D4C',
      gutterBorder: 'transparent',
      lineHighlight: '#00000012'
    }, settings),
    styles: [{
      tag: [_highlight.tags.meta, _highlight.tags.comment],
      color: '#804000'
    }, {
      tag: [_highlight.tags.keyword, _highlight.tags.strong],
      color: '#0000FF'
    }, {
      tag: [_highlight.tags.number],
      color: '#FF0080'
    }, {
      tag: [_highlight.tags.string],
      color: '#FF0080'
    }, {
      tag: [_highlight.tags.variableName],
      color: '#006600'
    }, {
      tag: [_highlight.tags.escape],
      color: '#33CC33'
    }, {
      tag: [_highlight.tags.tagName],
      color: '#1C02FF'
    }, {
      tag: [_highlight.tags.heading],
      color: '#0C07FF'
    }, {
      tag: [_highlight.tags.quote],
      color: '#000000'
    }, {
      tag: [_highlight.tags.list],
      color: '#B90690'
    }, {
      tag: [_highlight.tags.documentMeta],
      color: '#888888'
    }, {
      tag: [_highlight.tags["function"](_highlight.tags.variableName)],
      color: '#0000A2'
    }, {
      tag: [_highlight.tags.definition(_highlight.tags.typeName), _highlight.tags.typeName],
      color: '#6D79DE'
    }].concat((0, _toConsumableArray2["default"])(styles))
  });
};
exports.bbeditInit = bbeditInit;
var bbedit = bbeditInit();
exports.bbedit = bbedit;
