// @uiw/codemirror-theme-aura/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.auraInit = exports.aura = void 0;
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _highlight = require("@lezer/highlight");
var _codemirrorThemes = require("@uiw/codemirror-themes");
var auraInit = function auraInit(options) {
  var _ref = options || {},
    _ref$theme = _ref.theme,
    theme = _ref$theme === void 0 ? 'dark' : _ref$theme,
    _ref$settings = _ref.settings,
    settings = _ref$settings === void 0 ? {} : _ref$settings,
    _ref$styles = _ref.styles,
    styles = _ref$styles === void 0 ? [] : _ref$styles;
  return (0, _codemirrorThemes.createTheme)({
    theme: theme,
    settings: (0, _objectSpread2["default"])({
      background: '#21202e',
      foreground: '#edecee',
      caret: '#a277ff',
      selection: '#3d375e7f',
      selectionMatch: '#3d375e7f',
      gutterBackground: '#21202e',
      gutterForeground: '#edecee',
      gutterBorder: 'transparent',
      lineHighlight: '#a394f033'
    }, settings),
    styles: [{
      tag: _highlight.tags.keyword,
      color: '#a277ff'
    }, {
      tag: [_highlight.tags.name, _highlight.tags.deleted, _highlight.tags.character, _highlight.tags.macroName],
      color: '#edecee'
    }, {
      tag: [_highlight.tags.propertyName],
      color: '#ffca85'
    }, {
      tag: [_highlight.tags.processingInstruction, _highlight.tags.string, _highlight.tags.inserted, _highlight.tags.special(_highlight.tags.string)],
      color: '#61ffca'
    }, {
      tag: [_highlight.tags["function"](_highlight.tags.variableName), _highlight.tags.labelName],
      color: '#ffca85'
    }, {
      tag: [_highlight.tags.color, _highlight.tags.constant(_highlight.tags.name), _highlight.tags.standard(_highlight.tags.name)],
      color: '#61ffca'
    }, {
      tag: [_highlight.tags.definition(_highlight.tags.name), _highlight.tags.separator],
      color: '#edecee'
    }, {
      tag: [_highlight.tags.className],
      color: '#82e2ff'
    }, {
      tag: [_highlight.tags.number, _highlight.tags.changed, _highlight.tags.annotation, _highlight.tags.modifier, _highlight.tags.self, _highlight.tags.namespace],
      color: '#61ffca'
    }, {
      tag: [_highlight.tags.typeName],
      color: '#82e2ff'
    }, {
      tag: [_highlight.tags.operator, _highlight.tags.operatorKeyword],
      color: '#a277ff'
    }, {
      tag: [_highlight.tags.url, _highlight.tags.escape, _highlight.tags.regexp, _highlight.tags.link],
      color: '#61ffca'
    }, {
      tag: [_highlight.tags.meta, _highlight.tags.comment],
      color: '#6d6d6d'
    }, {
      tag: _highlight.tags.strong,
      fontWeight: 'bold'
    }, {
      tag: _highlight.tags.emphasis,
      fontStyle: 'italic'
    }, {
      tag: _highlight.tags.link,
      textDecoration: 'underline'
    }, {
      tag: _highlight.tags.heading,
      fontWeight: 'bold',
      color: '#a277ff'
    }, {
      tag: [_highlight.tags.atom, _highlight.tags.bool, _highlight.tags.special(_highlight.tags.variableName)],
      color: '#edecee'
    }, {
      tag: _highlight.tags.invalid,
      color: '#ff6767'
    }, {
      tag: _highlight.tags.strikethrough,
      textDecoration: 'line-through'
    }].concat((0, _toConsumableArray2["default"])(styles))
  });
};
exports.auraInit = auraInit;
var aura = auraInit();
exports.aura = aura;
