// @uiw/codemirror-theme-okaidia/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.okaidiaInit = exports.okaidia = void 0;
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _highlight = require("@lezer/highlight");
var _codemirrorThemes = require("@uiw/codemirror-themes");
var okaidiaInit = function okaidiaInit(options) {
  var _ref = options || {},
    _ref$theme = _ref.theme,
    theme = _ref$theme === void 0 ? 'dark' : _ref$theme,
    _ref$settings = _ref.settings,
    settings = _ref$settings === void 0 ? {} : _ref$settings,
    _ref$styles = _ref.styles,
    styles = _ref$styles === void 0 ? [] : _ref$styles;
  return (0, _codemirrorThemes.createTheme)({
    theme: theme,
    settings: (0, _objectSpread2["default"])({
      background: '#272822',
      foreground: '#FFFFFF',
      caret: '#FFFFFF',
      selection: '#49483E',
      selectionMatch: '#49483E',
      gutterBackground: '#272822',
      gutterForeground: '#FFFFFF70',
      lineHighlight: '#00000059'
    }, settings),
    styles: [{
      tag: [_highlight.tags.comment, _highlight.tags.documentMeta],
      color: '#8292a2'
    }, {
      tag: [_highlight.tags.number, _highlight.tags.bool, _highlight.tags["null"], _highlight.tags.atom],
      color: '#ae81ff'
    }, {
      tag: [_highlight.tags.attributeValue, _highlight.tags.className, _highlight.tags.name],
      color: '#e6db74'
    }, {
      tag: [_highlight.tags.propertyName, _highlight.tags.attributeName],
      color: '#a6e22e'
    }, {
      tag: [_highlight.tags.variableName],
      color: '#9effff'
    }, {
      tag: [_highlight.tags.squareBracket],
      color: '#bababa'
    }, {
      tag: [_highlight.tags.string, _highlight.tags.special(_highlight.tags.brace)],
      color: '#e6db74'
    }, {
      tag: [_highlight.tags.regexp, _highlight.tags.className, _highlight.tags.typeName, _highlight.tags.definition(_highlight.tags.typeName)],
      color: '#66d9ef'
    }, {
      tag: [_highlight.tags.definition(_highlight.tags.variableName), _highlight.tags.definition(_highlight.tags.propertyName), _highlight.tags["function"](_highlight.tags.variableName)],
      color: '#fd971f'
    },
    // { tag: t.keyword, color: '#f92672' },
    {
      tag: [_highlight.tags.keyword, _highlight.tags.definitionKeyword, _highlight.tags.modifier, _highlight.tags.tagName, _highlight.tags.angleBracket],
      color: '#f92672'
    }].concat((0, _toConsumableArray2["default"])(styles))
  });
};
exports.okaidiaInit = okaidiaInit;
var okaidia = okaidiaInit();
exports.okaidia = okaidia;
