// @uiw/codemirror-theme-bespin/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.bespinInit = exports.bespin = void 0;
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _highlight = require("@lezer/highlight");
var _codemirrorThemes = require("@uiw/codemirror-themes");
/**
 * @name Bespin
 * @author Mozilla / Jan T. Sott
 *
 * CodeMirror template by Jan T. Sott (https://github.com/idleberg/base16-codemirror)
 * Original Base16 color scheme by Chris Kempson (https://github.com/chriskempson/base16)
 */

var bespinInit = function bespinInit(options) {
  var _ref = options || {},
    _ref$theme = _ref.theme,
    theme = _ref$theme === void 0 ? 'dark' : _ref$theme,
    _ref$settings = _ref.settings,
    settings = _ref$settings === void 0 ? {} : _ref$settings,
    _ref$styles = _ref.styles,
    styles = _ref$styles === void 0 ? [] : _ref$styles;
  return (0, _codemirrorThemes.createTheme)({
    theme: theme,
    settings: (0, _objectSpread2["default"])({
      background: '#28211c',
      foreground: '#9d9b97',
      caret: '#797977',
      selection: '#36312e',
      selectionMatch: '#4f382b',
      gutterBackground: '#28211c',
      gutterForeground: '#666666',
      lineHighlight: 'rgba(255, 255, 255, 0.1)'
    }, settings),
    styles: [{
      tag: [_highlight.tags.atom, _highlight.tags.number, _highlight.tags.link, _highlight.tags.bool],
      color: '#9b859d'
    }, {
      tag: _highlight.tags.comment,
      color: '#937121'
    }, {
      tag: [_highlight.tags.keyword, _highlight.tags.tagName],
      color: '#cf6a4c'
    }, {
      tag: _highlight.tags.string,
      color: '#f9ee98'
    }, {
      tag: _highlight.tags.bracket,
      color: '#9d9b97'
    }, {
      tag: [_highlight.tags.variableName],
      color: '#5ea6ea'
    }, {
      tag: _highlight.tags.definition(_highlight.tags.variableName),
      color: '#cf7d34'
    }, {
      tag: [_highlight.tags["function"](_highlight.tags.variableName), _highlight.tags.className],
      color: '#cf7d34'
    }, {
      tag: [_highlight.tags.propertyName, _highlight.tags.attributeName],
      color: '#54be0d'
    }].concat((0, _toConsumableArray2["default"])(styles))
  });
};
exports.bespinInit = bespinInit;
var bespin = bespinInit();
exports.bespin = bespin;
