// @uiw/codemirror-theme-duotone/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.duotoneLightInit = exports.duotoneLight = exports.duotoneDarkInit = exports.duotoneDark = void 0;
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _highlight = require("@lezer/highlight");
var _codemirrorThemes = require("@uiw/codemirror-themes");
/**
 * @name duotone
 * @author Bram de Haan
 * by Bram de Haan, adapted from DuoTone themes by Simurai (http://simurai.com/projects/2016/01/01/duotone-themes)
 */

var duotoneLightInit = function duotoneLightInit(options) {
  var _ref = options || {},
    _ref$theme = _ref.theme,
    theme = _ref$theme === void 0 ? 'light' : _ref$theme,
    _ref$settings = _ref.settings,
    settings = _ref$settings === void 0 ? {} : _ref$settings,
    _ref$styles = _ref.styles,
    styles = _ref$styles === void 0 ? [] : _ref$styles;
  return (0, _codemirrorThemes.createTheme)({
    theme: theme,
    settings: (0, _objectSpread2["default"])({
      background: '#faf8f5',
      foreground: '#b29762',
      caret: '#93abdc',
      selection: '#e3dcce',
      selectionMatch: '#e3dcce',
      gutterBackground: '#faf8f5',
      gutterForeground: '#cdc4b1',
      gutterBorder: 'transparent',
      lineHighlight: '#EFEFEF'
    }, settings),
    styles: [{
      tag: [_highlight.tags.comment, _highlight.tags.bracket],
      color: '#b6ad9a'
    }, {
      tag: [_highlight.tags.atom, _highlight.tags.number, _highlight.tags.keyword, _highlight.tags.link, _highlight.tags.attributeName, _highlight.tags.quote],
      color: '#063289'
    }, {
      tag: [_highlight.tags.emphasis, _highlight.tags.heading, _highlight.tags.tagName, _highlight.tags.propertyName, _highlight.tags.variableName],
      color: '#2d2006'
    }, {
      tag: [_highlight.tags.typeName, _highlight.tags.url, _highlight.tags.string],
      color: '#896724'
    }, {
      tag: [_highlight.tags.operator, _highlight.tags.string],
      color: '#1659df'
    }, {
      tag: [_highlight.tags.propertyName],
      color: '#b29762'
    }, {
      tag: [_highlight.tags.unit, _highlight.tags.punctuation],
      color: '#063289'
    }].concat((0, _toConsumableArray2["default"])(styles))
  });
};
exports.duotoneLightInit = duotoneLightInit;
var duotoneLight = duotoneLightInit();
exports.duotoneLight = duotoneLight;
var duotoneDarkInit = function duotoneDarkInit(options) {
  var _ref2 = options || {},
    _ref2$theme = _ref2.theme,
    theme = _ref2$theme === void 0 ? 'light' : _ref2$theme,
    _ref2$settings = _ref2.settings,
    settings = _ref2$settings === void 0 ? {} : _ref2$settings,
    _ref2$styles = _ref2.styles,
    styles = _ref2$styles === void 0 ? [] : _ref2$styles;
  return (0, _codemirrorThemes.createTheme)({
    theme: theme,
    settings: (0, _objectSpread2["default"])({
      background: '#2a2734',
      foreground: '#6c6783',
      caret: '#ffad5c',
      selection: 'rgba(255, 255, 255, 0.1)',
      gutterBackground: '#2a2734',
      gutterForeground: '#545167',
      lineHighlight: '#36334280'
    }, settings),
    styles: [{
      tag: [_highlight.tags.comment, _highlight.tags.bracket],
      color: '#6c6783'
    }, {
      tag: [_highlight.tags.atom, _highlight.tags.number, _highlight.tags.keyword, _highlight.tags.link, _highlight.tags.attributeName, _highlight.tags.quote],
      color: '#ffcc99'
    }, {
      tag: [_highlight.tags.emphasis, _highlight.tags.heading, _highlight.tags.tagName, _highlight.tags.propertyName, _highlight.tags.className, _highlight.tags.variableName],
      color: '#eeebff'
    }, {
      tag: [_highlight.tags.typeName, _highlight.tags.url],
      color: '#7a63ee'
    }, {
      tag: _highlight.tags.operator,
      color: '#ffad5c'
    }, {
      tag: _highlight.tags.string,
      color: '#ffb870'
    }, {
      tag: [_highlight.tags.propertyName],
      color: '#9a86fd'
    }, {
      tag: [_highlight.tags.unit, _highlight.tags.punctuation],
      color: '#e09142'
    }].concat((0, _toConsumableArray2["default"])(styles))
  });
};
exports.duotoneDarkInit = duotoneDarkInit;
var duotoneDark = duotoneDarkInit();
exports.duotoneDark = duotoneDark;
